### PR TITLE
🏗 release: Update self-hosting support to use amp release

### PR DESCRIPTION
**Pull request by mdmower**
`amp release` copies static files and downloaded resources to supplement
an `amp dist` runtime. Support custom release flavor definitions and
update the [amp-framework-hosting documentation](https://github.com/ampproject/amphtml/blob/main/docs/spec/amp-framework-hosting.md).

### CODE REVIEW

File: build-system/tasks/release/index.js
Comment (danielrozenberg) [inline build-system/tasks/release/index.js:150]: Did you check whether `amp clean` removes the custom config? I don't believe `amp clean` removes files outside of `dist/`, `build/` etc... // cc: @rsimha

File: build-system/tasks/release/index.js
Comment (danielrozenberg) [inline build-system/tasks/release/index.js:514]: 1. The rest of this file uses `argv.flag_name` inside the various functions directly, instead of passing it between the function calls. Let's keep that consistent.
2. We tend to prefer `!!var` instead of `Boolean(var)`

File: build-system/tasks/release/index.js
Comment (danielrozenberg) [inline build-system/tasks/release/index.js:123]: Let's do the path.resolve here instead of inside the functions:

```suggestion
const CUSTOM_FLAVORS_CONFIG_PATH = path.resolve(
  __dirname,
  '../../../build-system/global-configs/custom-flavors-config.json'
);
const CUSTOM_OVERLAY_CONFIG_PATH = path.resolve(
  __dirname,
  '../../../build-system/global-configs/custom-config.json'
);
```

File: build-system/tasks/release/index.js
Comment (mdmower) [reply build-system/tasks/release/index.js:150]: Yes: https://github.com/ampproject/amphtml/blob/c8092fd78ae1b9c89450dcdbcaf71330dd206028/build-system/tasks/clean.js#L33

I would prefer that `build-system/global-configs/custom-config.json` and new config `build-system/global-configs/custom-flavor-configs.json` were not included in `amp clean`, but it's easy to argue both ways:
- These are not generated files; they are written by a user. So, in that sense, it makes sense to keep them around.
- It's easy to forget about them, and then you're left wondering why your builds don't behave like the AMP Project's builds.

Maybe a compromise would be to not delete by by default, but to emit a yellow warning that `amp clean` left them in place? This would simplify the current pull request..

File: build-system/tasks/release/index.js
Comment (mdmower) [reply build-system/tasks/release/index.js:123]: Unfortunately, `amp clean` does not support absolute paths.

File: build-system/tasks/release/index.js
Comment (mdmower) [reply build-system/tasks/release/index.js]: Item 1 addressed in commit fccb400.
Item 2 moot once item 1 was addressed.

File: build-system/tasks/release/index.js
Comment (danielrozenberg) [reply build-system/tasks/release/index.js]: I like the idea of `amp clean` warning users about config-altering files that it doesn't delete. @rsimha wdyt?

File: build-system/tasks/release/index.js
Comment (danielrozenberg) [reply build-system/tasks/release/index.js:123]: Oh I see, I missed that this gets used in two different places

File: build-system/tasks/release/index.js
Comment (rsimha) [reply build-system/tasks/release/index.js]: Just catching up with this thread. I couldn't have said it better than Matt: there are good reasons to clean the custom config file, and other good reasons to not clean it.

We already have a flag called `--include_subpackages` that optionally cleans all subpackage `node_modules` directories. Would it help to add a new flag `--include_custom_configs` that deletes any custom config files found locally? (This will address Matt's first point.)

And re: warning the developer when a custom config _is_ present, I think it should be printed by `amp build` and `amp dist` which are are more likely to be used than `amp clean`. WDYT?

Comment (mdmower) [reply build-system/tasks/release/index.js]: In commit 8bde827,
- `amp clean` emits: "Notice: Skipping build-system/global-configs/custom-config.json" (and similarly for `custom-flavor-configs.json`) if custom configs are detected
- `amp clean` supports new flag `--include_custom_configs` to also cleanup custom configs
- `amp dist`, `amp build`, `amp release` all emit notices similar to "Notice: prod config overlaid with custom-config.json" when a custom config overlay is applied
- `amp release` emits "Notice: release flavors supplemented by custom-flavors-config.json" when custom flavors are loaded

File: build-system/tasks/release/index.js
Comment (mdmower) [reply build-system/tasks/release/index.js]: In commit 8bde827, `amp clean` exclusions are no longer modified in `build-system/tasks/release/index.js`, so this change has been applied.

**Review by danielrozenberg — APPROVED** [review]
Approved. Let's get a 2nd review from @rsimha before merging this

**Review by rsimha — COMMENTED** [review]
Thanks for sending out this PR and for the attention to detail. Generally looks good modulo a few minor nits. Happy to review again once they are addressed.

File: build-system/tasks/clean.js
Comment (danielrozenberg) [reply build-system/tasks/clean.js:69]: @rsimha I think it's worth keeping the `Skipping paths:` part yellow, for emphasis

Comment (rsimha) [reply build-system/tasks/clean.js:69]: All I was aiming for was to roughly match this formatting:

image

No objections to any emphasis you think is useful, so feel free to pick what works. 👍

Comment (mdmower) [reply build-system/tasks/clean.js]: Sample outputs after commit de913d298b:
image

**Review by rsimha — APPROVED** [review]

File: build-system/tasks/clean.js
Comment (mdmower) [reply build-system/tasks/clean.js]: Suggestion applied in de913d298b (thought I do worry a bit that these paths will accidentally be added when someone follows the instructions at the top of the file list:
```
// Keep this list in sync with .gitignore, .eslintignore, and .prettierignore
```

File: build-system/global-configs/README.md
Comment (mdmower) [reply build-system/global-configs/README.md]: Suggestion applied in de913d2

File: build-system/tasks/clean.js
Comment (mdmower) [reply build-system/tasks/clean.js]: Suggestion applied in de913d2

File: docs/spec/amp-framework-hosting.md
Comment (mdmower) [reply docs/spec/amp-framework-hosting.md]: Suggestion applied in de913d2

**Comment by mdmower** [comment]
I appreciate the reviews @danielrozenberg and @rsimha. I think we're good here. I'll leave this open until 12pm PDT today (Oct 1) in case there's any last minute feedback before merging.

File: build-system/tasks/clean.js
Comment (rsimha) [reply build-system/tasks/clean.js]: Aha, I see your point. Since the config json files don't really belong in the "Output directories" section, perhaps we can create a new section called "Custom config files" in the three dotfiles, and add the comment about keeping them in sync at line 51?

**Comment by danielrozenberg** [comment]
No last minute feedback from me either, feel free to merge at your leisure!

File: build-system/tasks/clean.js
Comment (mdmower) [reply build-system/tasks/clean.js]: Good idea. I can do this small shuffle in a separate PR.

**Pull request merged**
